### PR TITLE
[AKS] Move AKSv2 monitoring function to operator

### PR DESCRIPTION
**Pull request by mjura**
In AKSv2 CheckLogAnalyticsWorkspaceForMonitoring has improved function
with updated metadate. This will be maintained in aks-operator.

Issue: https://github.com/rancher/rancher/issues/32967

### CODE REVIEW

**Review by cmurphy — COMMENTED** [review]
lgtm but this will need an aks-operator tag to update go.mod

**Review by rmweir — APPROVED** [review]
LGTM

**Review by thedadams — COMMENTED** [review]
Looks good except for new tag.

File: pkg/kontainer-engine/drivers/aks/aks_driver.go [hunk 1]
```diff
@@ -1433,11 +1307,3 @@ func (d *Driver) GetK8SCapabilities(ctx context.Context, _ *types.DriverOptions)
 	}, nil
 }
 
-func generateUniqueLogWorkspace(workspaceName string) string {
```
Comment (cbron) [inline pkg/kontainer-engine/drivers/aks/aks_driver.go:1436]: There is a test using this, make sure you are running tests locally. Did this test get merged to ask-operator ?

File: pkg/kontainer-engine/drivers/aks/aks_driver.go [hunk 2]
```diff
@@ -2,7 +2,6 @@ package aks
```
Comment (cbron) [inline pkg/kontainer-engine/drivers/aks/aks_driver.go:2]: failing go-fmt

**Review by cbron — CHANGES_REQUESTED** [review]

File: pkg/kontainer-engine/drivers/aks/aks_driver.go (hunk 2 above)
Comment (mjura) [reply pkg/kontainer-engine/drivers/aks/aks_driver.go:2]: Fixed

File: pkg/kontainer-engine/drivers/aks/aks_driver.go (hunk 1 above)
Comment (mjura) [reply pkg/kontainer-engine/drivers/aks/aks_driver.go:1436]: good catch, thank you I will move it there

Comment (mjura) [reply pkg/kontainer-engine/drivers/aks/aks_driver.go:1436]: It was added in https://github.com/rancher/aks-operator/pull/23

File: go.mod [hunk 3]
```diff
@@ -11,6 +11,8 @@ replace (
 	github.com/knative/pkg => github.com/rancher/pkg v0.0.0-20190514055449-b30ab9de040e
 	github.com/matryer/moq => github.com/rancher/moq v0.0.0-20200712062324-13d1f37d2d77
 	github.com/opencontainers/runc => github.com/opencontainers/runc v1.0.0-rc10
+	github.com/rancher/aks-operator => github.com/mjura/aks-operator v0.0.0-20210602074640-448d8ef7a09a
+	github.com/rancher/aks-operator/controller => github.com/mjura/aks-operator/controller v0.0.0-20210602074640-448d8ef7a09a
```
[the remark this answers is not in the document]
Comment (mjura) [reply go.mod]: fixed

**Review by cmurphy — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -133,7 +133,7 @@ require (
 	github.com/yvasiyarov/go-metrics v0.0.0-20150112132944-c25f46c4b940 // indirect
 	github.com/yvasiyarov/gorelic v0.0.7 // indirect
 	github.com/yvasiyarov/newrelic_platform_go v0.0.0-20160601141957-9c099fbc30e9 // indirect
-	golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2
+	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
```
Comment (cbron) [inline go.mod:136]: This must be because aks-operator bumped it, hopefully this is all compatible.

**Review by cbron — APPROVED** [review]

**Comment by cbron** [comment]
hard merging because its actually passing all tests

**Comment by mjura** [comment]
> hard merging because its actually passing all tests

@cbron  thank you for merging

**Pull request merged**